pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -1425,6 +1426,7 @@ function getResolvedPackage (
       engines: options.pkg.engines,
       os: options.pkg.os,
       libc: options.pkg.libc,
+      requiresBuild: !!options.pkg.scripts?.install ?? !!options.pkg.scripts?.preinstall ?? !!options.pkg.scripts?.postinstall,
```

Comment:
This won't work. This will always be false. We request the abbreviated metadata, which doesn't contain the `scripts` field. Even if we would request the full metadata, it wouldn't be enough to correctly detect if the package requires build. Some packages don't have install hooks in scripts but they have a `binding.gyp` file or a `hooks` directory. So in order to correctly set the property for each optional dependency, we'd have to download each optional dependency, which is not an option.

Reply:
Is the test I wrote flawed then? As in the test, it looks for a true and the test passes

Understand about the gyp/hooks etc but for just install scripts this seems to work